src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const resolvedPromise = new Promise((resolve) => {
+    element.addEventListener(eventName, function handleEvent() {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
```

Comment:
Consider removing the event listener after it has been triggered to prevent memory leaks and unnecessary event handling. You can use `element.removeEventListener(eventName, handleEvent);` within the `handleEvent` function.